core/network/fabric-interop-cc/contracts/interop/handle_external_request.go

```diff
@@ -104,16 +104,63 @@ func (s *SmartContract) HandleExternalRequest(ctx contractapi.TransactionContext
 	// 4. Calls application chaincode
 	arr := append([]string{viewAddress.CCFunc}, viewAddress.Args...)
 	byteArgs := strArrToBytesArr(arr)
-	pbResp := ctx.GetStub().InvokeChaincode(viewAddress.Contract, byteArgs, viewAddress.Channel)
-	if pbResp.Status != shim.OK {
-		errorMessage := fmt.Sprintf("Application chaincode invoke error: %s", string(pbResp.GetMessage()))
-		log.Error(errorMessage)
-		return "", errors.New(errorMessage)
+	
+	localCCId, err := wutils.GetLocalChaincodeID(ctx.GetStub())
+	payload := []byte("")
+	if localCCId == viewAddress.Contract {
+		// Interop call to InteropCC itself.
+		resp := ""
+		if viewAddress.CCFunc == "GetHTLCHash" {
+			if len(viewAddress.Args) > 2 {
+				errorMessage := fmt.Sprintf("Recieved more arguments than required 1 argument.")
+				log.Error(errorMessage)
+				return "", errors.New(errorMessage)
+			}
+			resp, err = s.GetHTLCHash(ctx, viewAddress.Args[0], viewAddress.Args[1])
+		} else if viewAddress.CCFunc == "GetHTLCHashByContractId" {
+			if len(viewAddress.Args) > 1 {
+				errorMessage := fmt.Sprintf("Recieved more arguments than required 1 argument.")
+				log.Error(errorMessage)
+				return "", errors.New(errorMessage)
+			}
+			resp, err = s.GetHTLCHashByContractId(ctx, viewAddress.Args[0])
+		} else if viewAddress.CCFunc == "GetHTLCHashPreImageByContractId" {
```

Comment:
Just to be consistent with the condition ordering in the above code, can you swap lines 127-133 with lines 134-140?